src/main/java/seedu/address/model/task/Task.java

```diff
@@ -78,7 +78,8 @@ public boolean equals(Object other) {
 
         Task otherTask = (Task) other;
         boolean bothHasTank = isTankRelatedTask() && otherTask.isTankRelatedTask();
-        boolean hasSameTank = bothHasTank && (tank.equals(otherTask.getTank()));
+        boolean bothNoTank = !isTankRelatedTask() && !otherTask.isTankRelatedTask();
+        boolean hasSameTank = (bothHasTank && (tank.equals(otherTask.getTank()))) || bothNoTank;
```

Comment:
Includes tasks with no tanks in duplication detection by detecting a task with no tank as being in the same tank as other tasks with no tank